### PR TITLE
feature: Add profile_name support for Feature Store ingestion

### PR DESCRIPTION
**Pull request by imingtsou**
*Issue #, if available:*

*Description of changes:*
Add `profile_name` for Feature Store ingestion so user can specify which credential to use for `PutRecord`.
*Testing done:*
Added one unit test and passed in Python 3.6/3.7/3.8.
I wrote a piece of sample ingestion with and without `profile_name` specified and both show the expected behavior.
## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [X] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [X] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [X] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [X] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [X] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [X] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [X] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [X] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [X] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/feature_store/feature_group.py [hunk 1]
```diff
@@ -607,6 +616,7 @@ def ingest(
             wait (bool): whether to wait for the ingestion to finish or not.
             timeout (Union[int, float]): ``concurrent.futures.TimeoutError`` will be raised
                 if timeout is reached.
+            profile_name (str): the profile credential should be used for ``PutRecord``.
```
Comment (alex-tang) [inline src/sagemaker/feature_store/feature_group.py:619]: Let's add more detail here. It should include 1/ what the default behavior is, if no profile name is specified and 2/ clarify that the profile credential is provided to a new boto3 Session, which is used initialize the sagemaker_featurestore_runtime client to ingest the data.

**Review by alex-tang — COMMENTED** [review]
Can you add more detail to the testing done section? I.e. describe the test cases did you performed.

**Review by alex-tang — DISMISSED** [review]

File: src/sagemaker/feature_store/feature_group.py [hunk 2]
```diff
@@ -599,6 +608,11 @@ def ingest(
         They can also be found from the IngestionManagerPandas' ``failed_rows`` function after
         the exception is thrown.
 
+        `profile_name` argument is an optional one. It will use the default credentialif None is
```
Comment (mufaddal-rohawala) [inline src/sagemaker/feature_store/feature_group.py:611]: a minor correction for adding space here: `credential if`

File: src/sagemaker/feature_store/feature_group.py [hunk 3]
```diff
@@ -190,6 +193,7 @@ def _ingest_single_batch(
                 client to perform boto calls.
             start_index (int): starting position to ingest in this batch.
             end_index (int): ending position to ingest in this batch.
+            profile_name (str): the profile credential should be used for ``PutRecord``.
```
Comment (mufaddal-rohawala) [inline src/sagemaker/feature_store/feature_group.py:196]: Lets add default values to new args in docstrings whereever possible for explicity.

**Review by mufaddal-rohawala — CHANGES_REQUESTED** [review]

File: src/sagemaker/feature_store/feature_group.py (hunk 2 above)
Comment (imingtsou) [reply src/sagemaker/feature_store/feature_group.py:611]: Sure.

File: src/sagemaker/feature_store/feature_group.py (hunk 3 above)
Comment (imingtsou) [reply src/sagemaker/feature_store/feature_group.py:196]: Sure.

File: .coveragerc [hunk 4]
```diff
@@ -1,4 +1,4 @@
 [run]
-concurrency = threading
+concurrency = thread
```
Comment (ahsan-z-khan) [inline .coveragerc:2]: Please change this to `threading` before merging.

**Review by ahsan-z-khan — CHANGES_REQUESTED** [review]

File: .coveragerc (hunk 4 above)
Comment (imingtsou) [reply .coveragerc]: Sure.

**Review by mufaddal-rohawala — APPROVED** [review]

**Review by ahsan-z-khan — APPROVED** [review]

**Pull request merged**